### ISSUE TITLE
[RFE] should we keep a static version of inih library in tree?

### ISSUE THREAD

**Issue by dougsland** [engelmi/hirte#430]
## Describe the bug

In the early days of the project it was added inih library (https://github.com/benhoyt/inih) in the source code. Should we keep this approach? I noticed `inih` released a new version last month. Can we use such library in a linking approach to we can keep getting recent bits?  

```
commit caec5cedb26b7cec0850346f5bf581757ac5bdb6
Author: Veronika Fuxova <vfuxova@redhat.com>
Date:   Thu Dec 1 13:56:39 2022 +0100

    Add inih lib to src tree

    Adding the whole lib now. The plan is to identify what we don't need and
    progressively get rid of it as we continue to refine our use case.
```

**Comment by mwperina** [comment]
So used the approach of copying the library because of simplicity :wink: 

But looking at the project, it's still quite active and, what's more important, it's packaged in [Fedora](https://koji.fedoraproject.org/koji/packageinfo?packageID=23025), it was even included in [EPEL 7/8](https://packages.fedoraproject.org/pkgs/inih/inih/) and older version is included in [AutoSD](https://autosd.sig.centos.org/AutoSD-9/nightly/repos/AutoSD/compose/AutoSD/x86_64/os/Packages/inih-49-6.el9.x86_64.rpm).

So if we would want to use the package instead of inlining it, we would need to:
1. Check if the compiled version aligns with our needs (many things in the project is configurable during the build only)
2. Add it to EPEL 9
3. Update the version in AutoSD
4. Remove inline code and adapt build process to use the library from a package instead

**Comment by rhatdan** [comment]
I would prefer to use the active upstream project rather then inline it, just so we can get the benefit of any security fixes found upstream. Should we get the library into RHEL though especially if we end up shipping hirte in RHEL?

**Comment by dougsland** [comment]
@engelmi solved that one via #432
